central/credentialexpiry/service/service_impl.go

```diff
@@ -153,14 +162,82 @@ func (s *serviceImpl) getScannerCertExpiry(ctx context.Context) (*v1.GetCertExpi
 			errorList.AddError(err)
 			// All the endpoints have failed.
 			if len(errorList.ErrorStrings()) == len(clairifyEndpoints) {
-				return nil, errors.New(errorList.String())
+				return nil, errorList.ToError()
 			}
 		case expiry := <-expiryC:
 			return &v1.GetCertExpiry_Response{Expiry: expiry}, nil
 		}
 	}
 }
 
+func (s *serviceImpl) getScannerV4CertExpiry(ctx context.Context) (*v1.GetCertExpiry_Response, error) {
+	if !features.ScannerV4.Enabled() {
+		return nil, errors.Wrap(errox.InvalidArgs, "Scanner V4 is not integrated")
+	}
+	indexerConfig := s.scannerConfigs[mtls.ScannerV4IndexerSubject]
+	matcherConfig := s.scannerConfigs[mtls.ScannerV4MatcherSubject]
+	if indexerConfig == nil && matcherConfig == nil {
+		return nil, errors.New("could not load TLS configs to talk to Scanner V4 indexer and matcher")
+	}
+
+	integration, exists, err := s.imageIntegrations.GetImageIntegration(ctx, iiStore.DefaultScannerV4Integration.GetId())
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to retrieve Scanner V4 image integration")
+	}
+	if !exists {
+		return nil, errors.New("Scanner V4 image integration missing")
+	}
+
+	s4Config := integration.GetScannerV4()
+	indexerEndpoint := scannerv4.DefaultIndexerEndpoint
+	if s4Config.GetIndexerEndpoint() != "" {
+		indexerEndpoint = s4Config.GetIndexerEndpoint()
+	}
+	matcherEndpoint := scannerv4.DefaultMatcherEndpoint
+	if s4Config.GetMatcherEndpoint() != "" {
+		matcherEndpoint = s4Config.GetMatcherEndpoint()
+	}
+
+	numEndpoints := 2
+	errC := make(chan error, numEndpoints)
+	expiryC := make(chan *types.Timestamp, numEndpoints)
+	getExpiry := func(subject mtls.Subject, endpoint string) {
+		expiry, err := s.getExpiryFunc(ctx, subject, s.scannerConfigs[subject], endpoint)
+		if err != nil {
+			errC <- err
+			return
+		}
+
+		log.Debugf("Obtained expiry for %q: %s", subject.Identifier, expiry)
+		expiryC <- expiry
+	}
+
+	go getExpiry(mtls.ScannerV4IndexerSubject, indexerEndpoint)
+	go getExpiry(mtls.ScannerV4MatcherSubject, matcherEndpoint)
```

Comment:
Why are we skipping the v4 DB?

Reply:
Scanner V4 DB is not reachable by Central due to `netpol`, and I don't believe it makes sense to remove that isolation for this purpose. This mirrors the current functionality with the existing Scanner.